java/src/main/java/gov/doc/ntia/sigmf/ext/global/algorithm/DataProduct.java

```diff
@@ -0,0 +1,30 @@
+package gov.doc.ntia.sigmf.ext.global.algorithm;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public abstract class DataProduct implements IDataProduct {
```

Comment:
Should also implement a `digital_filter` property, which will be inherited by the various data product classes.